src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/nonassociations/NonAssociationsResourceIntTest.kt

```diff
@@ -665,23 +673,96 @@ class NonAssociationsResourceIntTest : IntegrationTestBase() {
         .exchange()
         .expectStatus().isOk
         .expectBody()
-        .jsonPath("$[0].offenderNo").isEqualTo(offenderAtMoorlands.nomsId)
-        .jsonPath("$[0].nsOffenderNo").isEqualTo(offenderAtLeeds.nomsId)
-        .jsonPath("$[0].reason").isEqualTo("BUL")
-        .jsonPath("$[0].recipReason").isEqualTo("VIC")
-        .jsonPath("$[0].type").isEqualTo("LAND")
-        .jsonPath("$[0].authorisedBy").isEqualTo("Staff Member")
-        .jsonPath("$[0].effectiveDate").isEqualTo("2021-02-28")
-        .jsonPath("$[0].expiryDate").doesNotExist()
-        .jsonPath("$[0].comment").isEqualTo("this is a GET test!")
-        .jsonPath("$[1].offenderNo").doesNotExist()
+        .jsonPath("$.offenderNo").isEqualTo(offenderAtMoorlands.nomsId)
+        .jsonPath("$.nsOffenderNo").isEqualTo(offenderAtLeeds.nomsId)
+        .jsonPath("$.reason").isEqualTo("BUL")
+        .jsonPath("$.recipReason").isEqualTo("VIC")
+        .jsonPath("$.type").isEqualTo("LAND")
+        .jsonPath("$.authorisedBy").isEqualTo("Staff Member")
+        .jsonPath("$.effectiveDate").isEqualTo("2021-02-28")
+        .jsonPath("$.expiryDate").doesNotExist()
+        .jsonPath("$.comment").isEqualTo("this is a GET test!")
+    }
+
+    @Test
+    fun `offender not found`() {
+      webTestClient.get().uri("/non-associations/offender/A0000AA/ns-offender/A0000BB")
+        .headers(setAuthorisation(roles = listOf("ROLE_NOMIS_NON_ASSOCIATIONS")))
+        .exchange()
+        .expectStatus().isNotFound
+    }
+
+    @Test
+    fun `non-associations not found`() {
+      repository.deleteAllNonAssociations()
+      webTestClient.get().uri(
+        "/non-associations/offender/{offenderNo}/ns-offender/{nsOffenderNo}",
+        offenderAtLeeds.nomsId,
+        offenderAtMoorlands.nomsId,
+      )
+        .headers(setAuthorisation(roles = listOf("ROLE_NOMIS_NON_ASSOCIATIONS")))
+        .exchange()
+        .expectStatus().isNotFound
+    }
+
+    @Test
+    fun `NA exists but no open non-associations`() {
+      repository.deleteAllNonAssociations()
+      webTestClient.get().uri(
+        "/non-associations/offender/{offenderNo}/ns-offender/{nsOffenderNo}",
+        offenderAtMoorlands.nomsId,
+        offenderAtShrewsbury.nomsId,
+      )
+        .headers(setAuthorisation(roles = listOf("ROLE_NOMIS_NON_ASSOCIATIONS")))
+        .exchange()
+        .expectStatus().isNotFound
+    }
+
+    @Test
+    fun `get non-associations prevents access without appropriate role`() {
+      assertThat(
+        webTestClient.get()
+          .uri("/non-associations/offender/{offenderNo}/ns-offender/{nsOffenderNo}", "A1234AA", "A1234AA")
+          .headers(setAuthorisation(roles = listOf("ROLE_BLA")))
+          .exchange()
+          .expectStatus().isForbidden,
+      )
+    }
+  }
+
+  @Nested
+  inner class GetNonAssociationDetailsById {
```

Comment:
Very minor - (same for previous test) maybe rename inner classes to getOpenNonAssociationDetails and getAllNonAssociationDetails

Reply:
ok will sort that next time 👍